cookiecutter/schema.py

```diff
@@ -0,0 +1,198 @@
+        # min version of cookiecutter that is required by the template
+        "cookiecutter_version": {"type": "string"},
+        # python version constraints of the template
+        "python_requires": {"type": "string"},
+        # license of the template
+        "license": {"type": "string"},
+        # keywords that describe the goals of the template
+        "keywords": {"type": "array", "items": {"type": "string"}},
+        # the canonical url from where the template can be retrieved
+        "url": {"type": "string"},
+        # definition of the template's variables
+        "variables": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    # variable name (must be a valid python variable name)
+                    "name": {"type": "string"},
+                    # the default value for that variable
+                    "default": {},
+                    # text that will be displayed before the input field (keep it short!)
+                    "prompt": {"type": "string"},
+                    # more detailed description of this variable
+                    "description": {"type": "string"},
+                    # input data type (string, boolean, etc.)
+                    "type": {
+                        "type": "string",
+                        "enum": [
+                            "boolean",
+                            "yes_no",
+                            "int",
+                            "float",
+                            "uuid",
+                            "json",
+                            "string",
+                        ],
+                    },
+                    # validate user input with this regex
+                    "validation": {"type": "string"},
+                    # display this message if validation failed
+                    "validation_msg": {"type": "string"},
+                    # a list of items to choose from
+                    "choices": {"type": "array", "items": {"type": "string"}},
+                    # don't prompt the user for this variable, if this is set to false
+                    "prompt_user": {"type": "boolean"},
+                    # hide user input while typing (e.g. if you're asking for a password)
+                    "hide_input": {"type": "boolean"},
+                    # only show this prompt, if the specified condition is true
+                    "do_if": {"type": "string"},
+                    # skip this prompt, if the specified condition is true
+                    "skip_if": {"type": "string"},
+                    # skip to this variable name, if the user selected "no" in a "yes_no" prompt
+                    "if_no_skip_to": {"type": "string"},
+                    # skip to this variable name, if the user selected "yes" in a "yes_no" prompt
+                    "if_yes_skip_to": {"type": "string"},
+                    "validation_flags": {
```

Comment:
@louisdegaste can you add a docstring for this, and move it up to the other validation items?
`validation_flags` wasn't part of the original proposal, right? what is this about?

Reply:
validation_flags -> that one was part of the initial context v2 implementation that I pulled

Reply:
> @louisdegaste can you add a docstring for this, and move it up to the other validation items?

I don't get what you mean

Reply:
ah, those are for the regex engine... all right, that's fine